### ISSUE TITLE
Xcode console log duplicate timestamp

### ISSUE THREAD

**Issue by jacinzhang** [CocoaLumberjack/CocoaLumberjack#1251]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [x] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

1. My custom init
```objc
+ (void)logger {
    if (@available(iOS 10, *)) {
        [DDOSLogger.sharedInstance setLogFormatter:[ZXYLogFormatter new]];
        [DDLog addLogger:DDOSLogger.sharedInstance];
    } else {
        [DDTTYLogger.sharedInstance setLogFormatter:[ZXYLogFormatter new]];
        [DDLog addLogger:[DDTTYLogger sharedInstance]];
        [DDASLLogger.sharedInstance setLogFormatter:[ZXYLogFormatter new]];
        [DDLog addLogger:DDASLLogger.sharedInstance];
    }
    
    DDFileLogger *fileLogger = [[DDFileLogger alloc] init]; // File Logger
    fileLogger.rollingFrequency = 60 * 60 * 24; // 24 hour rolling
    fileLogger.logFileManager.maximumNumberOfLogFiles = 7;
    [fileLogger setLogFormatter:[ZXYLogFormatter new]];
    [DDLog addLogger:fileLogger];
}
```

```
@implementation ZXYLogFormatter

- (instancetype)init {
    self = [super init];
    if (self) {
        _dateFormatter = [[NSDateFormatter alloc] init];
        [_dateFormatter setDateFormat:@"yyyy-MM-dd HH:mm:ss.SSS ZZZ"];
        [_dateFormatter setCalendar:[[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian]];
    }
    return self;
}

- (NSString *)formatLogMessage:(DDLogMessage *)logMessage {
    NSString *timeStr = [_dateFormatter stringFromDate:logMessage -> _timestamp];
    NSString *flag = [self flagStrWithMessage:logMessage];
    NSString *fileName = [logMessage -> _file lastPathComponent];
    NSString *functionName = logMessage -> _function;
    NSInteger line = logMessage -> _line;
    NSString *message = logMessage -> _message;
    NSString *logString = [NSString stringWithFormat:@"%@ %@ %@:%@ %@: %@", timeStr, flag, fileName, @(line), functionName, message];
    return  logString;
}

- (NSString *)flagStrWithMessage:(DDLogMessage*)logMessage {
    NSString *logLevel;
    switch (logMessage->_flag) {
        case DDLogFlagError    : logLevel = @"E"; break;
        case DDLogFlagWarning  : logLevel = @"W"; break;
        case DDLogFlagInfo     : logLevel = @"I"; break;
        case DDLogFlagDebug    : logLevel = @"D"; break;
        default                : logLevel = @"V"; break;
    }
    return logLevel;
}

@end
```

And use it, It will duplicate timestamp log in Xcode console.

```
2021-09-20 17:08:21.443875+0800 Nuclear[67575:2542814] 2021-09-20 17:08:21.442 +0800 E ZXYTipsViewController.m:262 -[ZXYTipsViewController testLocaleString]: TÓMAS
```

Please fill in the detailed description of the issue (full textual output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Are you reporting a queue deadlock? If so, you must include a complete backtrace of all threads, which you can generate by typing `bt all` in the debugger after you hit the deadlock. Failure to do so will prevent us from helping you.

**Comment by ffried** [comment]
This is due to OSLog adding it's own timestamp. We can't do anything about this. You can prevent this by making your log formatter configurable and configure it to not add the timestamp when creating the log formatter to OSLogger.

**Comment by ffried** [comment]
I'm closing this since it's not really an issue with CocoaLumberjack.
